### DISCUSSION TITLE
Custom Taxonomies for products

### DISCUSSION THREAD

**Discussion by robruiz** [woocommerce/woocommerce#41474] [WooCommerce New Product Editor]
When I create a custom taxonomy with the standard register_taxonomy() function, the new block editor for products does not display it, so admins have no way of associating products with custom taxonomies currently without disabling the new editor. Once you disable the new editor and revert to the old UI, there the taxonomy displays and can be used. Hopefully this gets added to future releases where custom taxonomies show up in the new editor immediately after being registered.

**Comment by mattsherman** [comment]
Hello @robruiz!

Currently there is no automatic display of taxonomies in the new product editor. This is something we should explore in the future.

However, we do provide a generic `woocommerce/product-taxonomy-field` block that can be added to the product editor by an extension to allow for a custom taxonomy to be set.

https://github.com/woocommerce/woocommerce/blob/251812232d50acf18f8acd257d7e28d34138dfb5/packages/js/product-editor/src/blocks/generic/taxonomy/README.md

General product editor documentation, which includes details on how to extend the product editor:

https://github.com/woocommerce/woocommerce/tree/251812232d50acf18f8acd257d7e28d34138dfb5/docs/product-editor-development

We recognize our documentation is currently minimal. Please don't hesitate to follow up with any questions or comments!
